### DISCUSSION TITLE
Sync getStaticProps revalidation time and stale time of React Query (ISR, revalidate, staleTime)

### DISCUSSION THREAD

**Discussion by bennettdams** [trpc/trpc#2366] [Ideas]
Before I knew tRPC, I basically built my own SSG/hydration helpers (similar to [tRPC's ones](https://trpc.io/docs/ssg-helpers)). I encountered one thing doing that: The `getStaticProps` revalidation time and the query's stale time should be in sync, it should be the same value. 

Here's why:

https://twitter.com/BennettDams/status/1459184917192093700

This is also generally [mentioned in the docs](https://react-query-v3.tanstack.com/guides/ssr#staleness-is-measured-from-when-the-query-was-fetched-on-the-server), but is easily understandable with the `getStaticProps` example above.

---

I'm willing to create a PR for this, but I'm not sure how tRPC wants to tackle this. Maybe the query's stale time can be configured and what is returned from `createSSGHelpers` contains it, so one could is it directly in gSP's `revalidate` field. 

Regardless, attention should be drawn to this problem in the docs, right?

**Comment by sachinraja** [comment]
This does sound like a problem but I don't understand the solution. If the queries are cached and directly sent to the client, then `createSSGHelpers` will only run once. If you're saying to sync how long Next.js's revalidate is set for and how long the `staleTime` is, you can do this:
```ts
const revalidateSeconds = 60 * 5;

export async function getStaticProps() {
  const ssg = await createSSGHelpers({
    queryClientConfig: {
      // config...
      defaultOptions: {
        queries: {
          staleTime: 1000 * revalidateSeconds,
        },
      },
    },
  });

  // fetch stuff...

  return {
    props: {
      trpcState: ssg.dehydrate(),
    },
    revalidate: revalidateSeconds,
  };
}
```

**Reply by bennettdams** [reply]
>If the queries are cached and directly sent to the client, then createSSGHelpers will only run once.

I don't understand what you mean.

Regarding your code: Yes, you can do that easily, I just think that not so many people are aware of this, hence my question whether this should be added to the docs.

**Reply by sachinraja** [reply]
Isn't this more of a React Query issue though? We can link to their docs.
